### PR TITLE
Update and delete

### PR DESCRIPTION
**Pull request by donacrisante**

### CODE REVIEW

File: components/EntryList.js [hunk 1]
```diff
@@ -13,42 +14,68 @@ export default function EntryList({
...
+                {entry.date}, {entry.start} - {entry.destination}, {entry.km}km,{" "}
+                {entry.transport} {entry.fuel}, {entry.result} kg CO<sub>2</sub>
+              </p>
+              <button onClick={() => router.push(`/journey-list/${entry.id}`)}>
+                <span role="img" aria-label="A pencil">
+                  ✏️
+                </span>
+              </button>
+              <button
+                name="btn btn-light"
```
Comment (Kheirah) [inline components/EntryList.js:64]: Why do you need the name attribute?

File: components/EntryList.js [hunk 2]
```diff
@@ -13,42 +14,68 @@ export default function EntryList({
...
+            onClick={onShowFavouriteEntries}
+            isActive={filter === "favourites"}
+          >
+            Favourites{" "}
+            <Badge isActive={filter === "favourites"}>
+              {favouriteEntriesCount}
+            </Badge>
+          </Tab>
+        </Tabs>
+        <div name="journey-section__entries">
```
Comment (Kheirah) [inline components/EntryList.js:45]: What's the name attribute for?

File: components/EntryList.js [hunk 3]
```diff
@@ -13,42 +14,68 @@ export default function EntryList({
...
+  const router = useRouter();
+
+  function handleBackToForm() {
+    router.back();
+  }
 
   return (
     <>
       <h2>Journeys</h2>
       <section name="journeys-section">
```
Comment (Kheirah) [inline components/EntryList.js:29]: You have quite a few name attributes which you don't need.

File: pages/journey-list/index.js [hunk 4]
```diff
@@ -10,6 +10,10 @@ export default function JourneyList({
   result,
   id,
   onToggleFavourite,
+  selectedEntry,
+  setSelectedEntry,
```
Comment (Kheirah) [inline pages/journey-list/index.js:14]: Where do `selectedEntry` and `setSelectedEntry` come from?
No such props are passed down in `_app.js`. Remove them here.

File: pages/journey-list/index.js [hunk 5]
```diff
@@ -23,6 +27,10 @@ export default function JourneyList({
       result={result}
       id={id}
       onToggleFavourite={onToggleFavourite}
+      selectedEntry={selectedEntry}
+      setSelectedEntry={setSelectedEntry}
```
Comment (Kheirah) [inline pages/journey-list/index.js:31]: Remove them here as well since they're all `undefined`.

File: pages/_app.js [hunk 6]
```diff
@@ -22,9 +23,33 @@ export default function App({ Component, pageProps }) {
...
+    setEntries(
+      entries.map((entry) =>
+        entry.id === updatedEntry.id ? { ...entry, ...updatedEntry } : entry
+      )
+    );
+    router.back();
+    alert("Journey updated successfully!");
+  }
+
+  function handleDelete(selectedEntry) {
```
Comment (Kheirah) [inline pages/_app.js:41]: The name `selectedEntry` is confusing since it's not an entry; it's an id.

**Review by Kheirah — CHANGES_REQUESTED** [review]
Well done!

Have a look at the comments and remove things you no longer need. It's mostly residual code that needs to go.

**Review by MerleMak — APPROVED** [review]
Looks good 👍🏼

**Pull request merged**